api/allocations.go

```diff
@@ -79,6 +79,17 @@ func (a *Allocations) GC(alloc *Allocation, q *QueryOptions) error {
 	return err
 }
 
+func (a *Allocations) Restart(alloc *Allocation, taskName string, q *QueryOptions) error {
+	// TODO(dani): We can't depend on nomad/structs here any more, and I don't want
+	//             to duplicate a single use definition. When we move all of Nomad
+	//             to modules, we should fix this.
```

Comment:
A single use struct would be ~1 more line of code than this comment, so let's just make a single use struct eg [AllocatedSharedResources](https://github.com/hashicorp/nomad/blob/e4293eefab693cd32554d1ce388574e54a0abf87/api/allocations.go#L202)